src/crimson/os/seastore/async_cleaner.cc

```diff
@@ -646,46 +666,55 @@ JournalTrimmerImpl::trim_dirty()
 
   auto& shard_stats = extent_callback->get_shard_stats();
   ++(shard_stats.trim_dirty_num);
-  ++(shard_stats.pending_bg_num);
 
-  return repeat_eagain([this, FNAME, &shard_stats] {
-    ++(shard_stats.repeat_trim_dirty_num);
+  auto target = get_dirty_tail_stop_target();
+  return ::crimson::repeat([this, FNAME, &shard_stats, target] {
```

Comment:
Looks like this additional loop can violate the EPM reservation mechanism that it expects each round of `trim_dirty()` cannot exceed `rewrite_dirty_bytes_per_cycle`.